tools/c7n_azure/c7n_azure/resources/sqlserver.py

```diff
@@ -23,3 +33,89 @@ class resource_type(ArmResourceManager.resource_type):
         service = 'azure.mgmt.sql'
         client = 'SqlManagementClient'
         enum_spec = ('servers', 'list', None)
+
+
+@SqlServer.filter_registry.register('firewall-rules')
+class SqlServerFirewallRulesFilter(Filter):
+    """Filters SQL servers by the firewall rules
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: servers-with-firewall
+                resource: azure.sqlserver
+                filters:
+                  - type: firewall-rules
+                    include:
+                    - '131.107.160.2-131.107.160.3'
+                    - 10.20.20.0/24
+    """
+
+    schema = type_schema(
+        'firewall-rules',
+        **{
+            'include': {'type': 'array', 'items': {'type': 'string'}},
+            'equal': {'type': 'array', 'items': {'type': 'string'}}
+        })
+
+    def __init__(self, data, manager=None):
+        super(SqlServerFirewallRulesFilter, self).__init__(data, manager)
+        self.policy_include = None
+        self.policy_equal = None
+        self.client = self.manager.get_client()
+
+    def validate(self):
+        self.policy_include = IpRangeHelper.parse_ip_ranges(self.data, 'include')
+        self.policy_equal = IpRangeHelper.parse_ip_ranges(self.data, 'equal')
+
+        has_include = self.policy_include is not None
+        has_equal = self.policy_equal is not None
+
+        if has_include and has_equal:
+            raise FilterValidationError('Cannot have both include and equal.')
+
+        if not has_include and not has_equal:
+            raise FilterValidationError('Must have either include or equal.')
+
+        return True
+
+    def process(self, resources, event=None):
+        result, _ = ThreadHelper.execute_in_parallel(
+            resources=resources,
+            event=event,
+            execution_method=self._check_resources,
+            executor_factory=self.executor_factory,
+            log=log,
+            max_workers=constants.DEFAULT_MAX_THREAD_WORKERS,
```

Comment:
nit: ThreadHelper uses these constants as a default, so we can remove it from here.